src/SharedAcceptanceTests.All/ConnectionStringHelper.cs

```diff
@@ -40,10 +40,12 @@ static string GetEnvironmentVariable(string variable)
         }
 
         // the SDK uses exactly this method of changing the underlying executor
-        public static bool IsPremiumEndpoint(CloudTableClient cloudTableClient)
+        public static bool IsPremiumEndpoint(TableServiceClient cloudTableClient)
         {
-            var lowerInvariant = cloudTableClient.StorageUri.PrimaryUri.OriginalString.ToLowerInvariant();
-            return (lowerInvariant.Contains("https://localhost") && cloudTableClient.StorageUri.PrimaryUri.Port != 10002) || lowerInvariant.Contains(".table.cosmosdb.") || lowerInvariant.Contains(".table.cosmos.");
+            // TODO: figure out how to verify whether it's a premium endpoint or not
```

Comment:
Maybe we can steal this from https://github.com/Particular/NServiceBus.AzureStorageQueues/pull/895/files#diff-d0295f7348a1657628689f773bbaf0aaeab2cd2e8e37ef4af37e5557d0fa9b05R63 ?